GDJS/Runtime/runtimewatermark.ts

```diff
@@ -0,0 +1,427 @@
+          this._svgElement.setAttribute('width', this._logoWidth.toString());
+        }
+        this.updateBackgroundHeight(height);
+        if (this._watermarkBackgroundElement) {
+          this._watermarkBackgroundElement.style.height = `${this._backgroundHeight}px`;
+        }
+        this.updateMargin(height);
+        if (this._watermarkContainerElement) {
+          this.updateElementMargins(this._watermarkContainerElement);
+        }
+      }
+
+      private addWatermarkToGameContainer(container: HTMLElement) {
+        const gameContainerRectangle = container.getBoundingClientRect();
+        this.updateFontSize(gameContainerRectangle.height);
+        this.updateLogoSize(gameContainerRectangle.height);
+        this.updateBackgroundHeight(gameContainerRectangle.height);
+
+        this._watermarkContainerElement = this.createDivContainer();
+        this.createBackground();
+        const textContainer = document.createElement('div');
+
+        this.generateSVGLogo(gameContainerRectangle.height);
+        this.createMadeWithTextElement();
+        this.createUsernameTextElement();
+        if (this._svgElement)
+          this._watermarkContainerElement.appendChild(this._svgElement);
+        if (this._madeWithTextElement)
+          textContainer.appendChild(this._madeWithTextElement);
+        if (this._usernameTextElement)
+          textContainer.appendChild(this._usernameTextElement);
+        this._watermarkContainerElement.appendChild(textContainer);
+        addTouchAndClickEventListeners(
+          this._watermarkContainerElement,
+          this.openCreatorProfile.bind(this)
+        );
+        if (this._watermarkBackgroundElement)
+          container.appendChild(this._watermarkBackgroundElement);
+
+        container.appendChild(this._watermarkContainerElement);
+
+        this.setupAnimations();
+      }
+
+      private createBackground() {
+        this._watermarkBackgroundElement = document.createElement('div');
+        this._watermarkBackgroundElement.setAttribute(
+          'id',
+          'watermark-background'
+        );
+        this._watermarkBackgroundElement.style.height = `${this._backgroundHeight}px`;
+        this._watermarkBackgroundElement.style.opacity = '0';
+        if (this._placement.startsWith('top')) {
+          this._watermarkBackgroundElement.style.top = '0';
+          this._watermarkBackgroundElement.style.backgroundImage =
+            'linear-gradient(180deg, rgba(38, 38, 38, .6) 0%, rgba(38, 38, 38, 0) 100% )';
+        } else {
+          this._watermarkBackgroundElement.style.bottom = '0';
+          this._watermarkBackgroundElement.style.backgroundImage =
+            'linear-gradient(0deg, rgba(38, 38, 38, .6) 0%, rgba(38, 38, 38, 0) 100% )';
+        }
+      }
+
+      private setupAnimations() {
+        // Necessary to trigger fade in transition
+        requestAnimationFrame(() => {
+          // Display the watermark
+          setTimeout(() => {
```

Comment:
Idea: you could almost have used an async function with `await delay(xxxx);` between each part, with something like: `const delay = (time) => new Promise(resolve => setTimeout(resolve, time))`

Reply:
That's true. Do you think I should make the change?

Reply:
I'm unsure, depends if you think there is a risk that we want to run something and note something else (but in this case, the `if` works!) or if we want to *always* do something after a certain time (but in this case you can adapt the delay or just use a regular setTimeout).